src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -227,17 +227,16 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
 	<!-- Chooses the native Android debug server, valid values are: gdbserver, llgs and ds2 -->
 	<AndroidGdbDebugServer Condition="'$(AndroidGdbDebugServer)' == ''">Gdb</AndroidGdbDebugServer>
 
-    <!--
-    <_LibraryProjectImportsDirectoryName Condition="'$(UseShortFileNames)' == 'true'">jlibs</_LibraryProjectImportsDirectoryName>
-    <_LibraryProjectImportsDirectoryName Condition="'$(UseShortFileNames)' != 'true'">library_project_imports</_LibraryProjectImportsDirectoryName>
-	<_NativeLibraryImportsDirectoryName Condition="'$(UseShortFileNames)' == 'true'">libs</_NativeLibraryImportsDirectoryName>
-	<_NativeLibraryImportsDirectoryName Condition="'$(UseShortFileNames)' != 'true'">native_library_imports</_NativeLibraryImportsDirectoryName>
-    -->
-    <_LibraryProjectImportsDirectoryName>library_project_imports</_LibraryProjectImportsDirectoryName>
-	<_NativeLibraryImportsDirectoryName>native_library_imports</_NativeLibraryImportsDirectoryName>
+	<_LibraryProjectImportsDirectoryName Condition=" '$(_LibraryProjectImportsDirectoryName)'=='' And '$(UseShortFileNames)' == 'True' ">jl</_LibraryProjectImportsDirectoryName>
+	<_NativeLibraryImportsDirectoryName Condition=" '$(_NativeLibraryImportsDirectoryName)'=='' And '$(UseShortFileNames)' == 'True' ">nl</_NativeLibraryImportsDirectoryName>
+
+	<_LibraryProjectImportsDirectoryName Condition=" '$(_LibraryProjectImportsDirectoryName)'==''">library_project_imports</_LibraryProjectImportsDirectoryName>
+	<_NativeLibraryImportsDirectoryName Condition=" '$(_NativeLibraryImportsDirectoryName)'==''">native_library_imports</_NativeLibraryImportsDirectoryName>
 	<_AndroidResourcePathsCache>$(IntermediateOutputPath)resourcepaths.cache</_AndroidResourcePathsCache>
 	<_AndroidLibraryImportsCache>$(IntermediateOutputPath)libraryimports.cache</_AndroidLibraryImportsCache>
 	<_AndroidLibraryProjectImportsCache>$(IntermediateOutputPath)libraryprojectimports.cache</_AndroidLibraryProjectImportsCache>
+	<_AndroidLibrayProjectIntermediatePath Condition=" '$(_AndroidLibrayProjectIntermediatePath)' == '' And '$(UseShortFileNames)' == 'True' ">$(IntermediateOutputPath)lp</_AndroidLibrayProjectIntermediatePath>
```

Comment:
Not sure if this is important/matters, but e.g. in the previous `<ResolveLibraryProjectImports/>` invocation, we had:

```xml
<ResolveLibraryProjectImports
        OutputImportDirectory="$(IntermediateOutputPath)__library_projects__\"
```

That would now be:

```xml
<ResolveLibraryProjectImports
        OutputImportDirectory="$(IntermediateOutputPath)__library_projects__"
```

Note the *removal* of the trailing `\`.

*Is this a problem*? Possibly; I know we've encountered previous bugs where the presence/lack of a trailing `/` would alter behavior, causing the "wrong" directory to be deleted. For example, monodroid/a22d699b / [Bug #27442](https://bugzilla.xamarin.com/show_bug.cgi?id=27442), which references the "same" property!

For sanity, this property should also include a trailing `\`, both on this line and the following line.